### PR TITLE
content tab group

### PR DESCRIPTION
**Pull request by alanrenmsft**
as per the mock up of new dashboard, we need databases tab under contents group of server dashboard. removed the databases widget from the home tab.

https://www.figma.com/proto/QRkuP8YG7sZr87LRoS210ldo/ADS_Framework_v0.03?node-id=3058%3A97242&scaling=min-zoom

Screen Shot 2020-01-14 at 5 40 44 PM

### CODE REVIEW

File: src/sql/workbench/contrib/dashboard/browser/pages/serverDashboardPage.component.ts [hunk 1]
```diff
@@ -68,4 +68,32 @@ export class ServerDashboardPage extends DashboardPage implements OnInit {
...
+	getContentTabs(): TabConfig[] {
+		return [{
+			id: 'databasesTab',
+			provider: anyProviderName,
+			publisher: undefined,
+			title: nls.localize('databasesTabName', "Databases"),
+			container: {
+				'widgets-container': [
+					{
+						name: nls.localize('explorerWidgetTitle', "Search"),
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/dashboard/browser/pages/serverDashboardPage.component.ts]: Why is this being moved in here? Ideally all this config should be contributed from the JSON

File: src/sql/workbench/contrib/dashboard/browser/core/dashboardPage.component.ts [hunk 2]
```diff
@@ -172,6 +174,32 @@ export abstract class DashboardPage extends AngularDisposable implements IConfig
 		}));
 	}
 
+	private addContentTabGroup(): void {
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/dashboard/browser/core/dashboardPage.component.ts]: Why not just have the tab groups be defined in the JSON instead? If we want to add more groups this just gets messier and messier. And being able to define any arbitrary amount of tab groups seems like the direction we want to head anyways (and it's just an implementation that the dashboard for now is going to have two groups - contents and extensions)

Comment (alanrenmsft) [reply src/sql/workbench/contrib/dashboard/browser/core/dashboardPage.component.ts]: good point, i have made it more extensible with the latest commit.

File: extensions/mssql/package.json [hunk 3]
```diff
@@ -485,6 +491,28 @@
             }
           ]
         }
+      },
+      {
+        "id": "databases-tab",
+        "description": "%databases-tab-description%",
+        "provider": "MSSQL",
+        "title": "%databases-tab-title%",
+        "when": "",
```
Comment (alanrenmsft) [inline extensions/mssql/package.json]: I need to figure out the when expression for it to only show for server context. i will have another commit.

Comment (alanrenmsft) [reply extensions/mssql/package.json]: fixed

File: src/sql/workbench/contrib/dashboard/browser/core/dashboardTab.contribution.ts [hunk 4]
```diff
@@ -157,3 +156,57 @@ ExtensionsRegistry.registerExtensionPoint<IDashboardTabContrib | IDashboardTabCo
...
+		{
+			type: 'array',
+			items: tabGroupSchema
+		}
+	]
+};
+
+ExtensionsRegistry.registerExtensionPoint<IDashboardTabContrib | IDashboardTabContrib[]>({ extensionPoint: 'dashboard.tabGroups', jsonSchema: tabGroupContributionSchema }).setHandler(extensions => {
+
+	function handleCommand(tabgroup: IDashboardTabGroupContrib, extension: IExtensionPointUser<any>) {
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/dashboard/browser/core/dashboardTab.contribution.ts]: Why is this named handleCommand? Doesn't seem to match what you're actually doing (handling tabGroups)

Comment (alanrenmsft) [reply src/sql/workbench/contrib/dashboard/browser/core/dashboardTab.contribution.ts]: :) updated

**Comment by alanrenmsft** [comment]
merged without pushing the commit, i am going to have a separate PR to update the method name with related changes.

**Pull request merged**